src/MetaMorphoFactory.sol

```diff
@@ -47,8 +50,9 @@ contract MetaMorphoFactory {
         string memory name,
         string memory symbol,
         bytes32 salt
-    ) external returns (MetaMorpho metaMorpho) {
-        metaMorpho = new MetaMorpho{salt: salt}(initialOwner, MORPHO, initialTimelock, asset, name, symbol);
+    ) external returns (IMetaMorpho metaMorpho) {
+        metaMorpho =
+            IMetaMorpho(address(new MetaMorpho{salt: salt}(initialOwner, MORPHO, initialTimelock, asset, name, symbol)));
```

Comment:
```suggestion
    ) external returns (IMetaMorphoStaticTyping metaMorpho) {
        metaMorpho = new MetaMorpho{salt: salt}(initialOwner, MORPHO, initialTimelock, asset, name, symbol);
```

Reply:
Yes, also possible indeed
I preferred the one I suggested because it is the interface expected to be used by integrators